python/pyiceberg/expressions/literals.py

```diff
@@ -376,6 +407,12 @@ class DecimalLiteral(Literal[Decimal]):
     def __init__(self, value: Decimal):
         super().__init__(value, Decimal)
 
+    def increment(self) -> DecimalLiteral:
+        return DecimalLiteral(self.value + 1)
```

Comment:
We actually want to increment by the smallest amount possible, since what is actually happening is adjusting the boundary to closed rather than actually incrementing. That is, `literal("12.34").to(DecimalType(9, 2).increment())` should result in `12.35`, not `13.35`.

We do this in Java by converting to the unscaled value, incrementing, and then converting back. I think we should do the same thing here:

```
def increment_unscaled(d: Decimal):
    (_, _, original_scale) = d.to_tuple()
    unscaled = decimal_to_unscaled(self.value) + 1
    return unscaled_to_decimal(unscaled, original_scale)
```

Reply:
How do you feel about:
```python
    def increment(self) -> Literal[Decimal]:
        inc_value = 1 / (10 ** abs(self.value.as_tuple().exponent))
        return DecimalLiteral(self.value + Decimal(str(inc_value)))

    def decrement(self) -> Literal[Decimal]:
        inc_value = 1 / (10 ** abs(self.value.as_tuple().exponent))
        return DecimalLiteral(self.value - Decimal(str(inc_value)))
```
Added tests as well:
```python
def test_decimal_literal_increment():
    dec = DecimalLiteral(Decimal('10.123'))
    # Twice to check that we don't mutate the value
    assert dec.increment() == DecimalLiteral(Decimal('10.124'))
    assert dec.increment() == DecimalLiteral(Decimal('10.124'))
    # To check that the scale is still the same
    assert dec.increment().value.as_tuple() == Decimal('10.124').as_tuple()

def test_decimal_literal_dencrement():
    dec = DecimalLiteral(Decimal('10.123'))
    # Twice to check that we don't mutate the value
    assert dec.decrement() == DecimalLiteral(Decimal('10.122'))
    assert dec.decrement() == DecimalLiteral(Decimal('10.122'))
    # To check that the scale is still the same
    assert dec.decrement().value.as_tuple() == Decimal('10.122').as_tuple()
```